src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -135,7 +135,28 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
                                         user,
                                         ActionListener.wrap(stateResponse -> {
                                             logger.info("create state workflow doc");
-                                            listener.onResponse(new WorkflowResponse(globalContextResponse.getId()));
+                                            if (request.isProvision()) {
+                                                logger.info("provision parameter");
```

Comment:
nit: can you change to "provisioning workflow after creation" or "provisioning param is set to true", not 100% how clear `provision parameter` is.